packages/twenty-server/src/engine/metadata-modules/remote-server/dtos/update-remote-server.input.ts

```diff
@@ -0,0 +1,27 @@
+import { Field, InputType } from '@nestjs/graphql';
+
+import { IsOptional } from 'class-validator';
+import GraphQLJSON from 'graphql-type-json';
+
+import {
+  ForeignDataWrapperOptions,
+  RemoteServerType,
+  UserMappingOptions,
+} from 'src/engine/metadata-modules/remote-server/remote-server.entity';
+
+@InputType()
+export class UpdateRemoteServerInput<T extends RemoteServerType> {
+  @Field(() => String)
+  id: T;
+
+  @Field(() => String)
+  foreignDataWrapperType: T;
+
+  @IsOptional()
+  @Field(() => GraphQLJSON)
+  foreignDataWrapperOptions: ForeignDataWrapperOptions<T>;
+
+  @IsOptional()
+  @Field(() => GraphQLJSON, { nullable: true })
+  userMappingOptions?: UserMappingOptions;
```

Comment:
Let's add some form of validation on this json type directly from graphql by creating an input type
```
@InputType()
class UserMappingOptionsInput {
  @Field(() => String)
  @IsString()
  username: string;

  @Field(() => String)
  @IsString()
  password: string;
}
```
or something like that, then
```
  @IsOptional()
  @Field(() => UserMappingOptionsInput, { nullable: true })
  userMappingOptions?: UserMappingOptionsInput;
```

If you don't want a new input type, you can also use class-validator and validate your json in your resolver like
```
class UserMappingOptionsValidator {
  @IsDefined()
  @IsString()
  username: string;

  @IsDefined()
  @IsString()
  password: string;
}
```
and then use validateOrReject from 'class-validator' package.
What do you think?